### PR TITLE
Clarify some of the language around determinism

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-compiler can u please review this doc update?

File: docs/compilers/Deterministic Inputs.md [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-We are aiming to make the compilers ultimately deterministic (https://github.com/dotnet/roslyn/issues/372). What that means is that the "same inputs" will cause the compilers to produce the "same outputs". 
+The C# and VB compilers are fully determinstic when the `/deterministic` option is specified (this is the default in the .NET SDK). This means that the "same inputs" will cause the compilers to produce the "same outputs" with byte for byte equivalence. 
 
 The following are considered inputs to the compiler for the purpose of determinism:
 
-- The sequence of command-line parameters
-- The contents of the compiler's `.rsp` response file.
-- The precise version of the compiler used, and its referenced assemblies
+- The sequence of command-line parameters (order is important)
+- The precise version of the compiler used and the files included in it's deployment: reference assemblies, rsp, etc ...
```
Comment (cston) [inline docs/compilers/Deterministic Inputs.md:9]: >it's [](start = 69, length = 4)

"its"

File: docs/compilers/Deterministic Inputs.md [hunk 2]
```diff
@@ -17,15 +16,17 @@ The following are considered inputs to the compiler for the purpose of determini
   - the strong name key file
   - `@` response files
   - Analyzers
+  - Generators
   - Rulesets
-  - "additional files" that may be used by analyzers
-- The current culture (for the language in which diagnostics and exception messages are produced).
+  - "additional files" that may be used by analyzers and generators
+- The current culture if `/preferuilang` is not specified (for the language in which diagnostics and exception messages are produced).
```
Comment (cston) [inline docs/compilers/Deterministic Inputs.md:22]: >`/preferuilang` [](start = 25, length = 15)

`/preferreduilang`

File: docs/compilers/Deterministic Inputs.md [hunk 3]
```diff
@@ -17,15 +16,17 @@ The following are considered inputs to the compiler for the purpose of determini
...
 - The current OS code page if `/codepage` is not specified and any of the input source files do not have BOM and are not UTF-8 encoded.
 - The existence, non-existence, and contents of files on the compiler's search paths (specified, e.g. by `/lib` or `/recurse`)
 - The CLR platform on which the compiler is run:
   - The result of `double` arithmetic performed for constant-folding may use excess precision on some platforms.
   - The compiler uses Unicode tables provided by the platform.
 - The version of the zlib library that the CLR uses to implement compression (when `/embed` or `/debug:embedded` is used).
-- The value of `%LIBPATH%`, as it can affect analyzer dependency loading.
+- The value of `%LIBPATH%`, as it can affect reference discovery if not fully qualified and analyzer / generator dependency loading.
```
Comment (cston) [inline docs/compilers/Deterministic Inputs.md:29]: > if not fully qualified [](start = 64, length = 23)

This clause makes it difficult to parse the subsequent "and analyzer / generator dependency loading". Perhaps add parentheses: "(if not fully qualified)".

File: docs/compilers/Deterministic Inputs.md [hunk 4]
```diff
@@ -1,13 +1,12 @@
 Deterministic Inputs
 ====================
 
-We are aiming to make the compilers ultimately deterministic (https://github.com/dotnet/roslyn/issues/372). What that means is that the "same inputs" will cause the compilers to produce the "same outputs". 
+The C# and VB compilers are fully determinstic when the `/deterministic` option is specified (this is the default in the .NET SDK). This means that the "same inputs" will cause the compilers to produce the "same outputs" with byte for byte equivalence.
```
Comment (jcouv) [inline docs/compilers/Deterministic Inputs.md:4]:
```suggestion
The C# and VB compilers are fully determinstic when the `/deterministic` option is specified (this is the default in the .NET SDK). This means that the "same inputs" will cause the compilers to produce the "same outputs" byte for byte. 
```

Comment (Therzok) [reply docs/compilers/Deterministic Inputs.md:4]: To add on @jcouv 's suggestion: 
```suggestion
The C# and VB compilers are fully deterministic when the `/deterministic` option is specified (this is the default in the .NET SDK). This means that the "same inputs" will cause the compilers to produce the "same outputs" byte for byte. 
```

File: docs/compilers/Deterministic Inputs.md (hunk 3 above)
Comment (jaredpar) [reply docs/compilers/Deterministic Inputs.md]: I clarified the language here. The generator / analyzer dependency isn't about whether or not they're fully qualified but rather how the runtime deals with analyzer / generator dependency loading.

File: docs/compilers/Deterministic Inputs.md [hunk 5]
```diff
@@ -1,13 +1,12 @@
 Deterministic Inputs
 ====================
 
-We are aiming to make the compilers ultimately deterministic (https://github.com/dotnet/roslyn/issues/372). What that means is that the "same inputs" will cause the compilers to produce the "same outputs". 
+The C# and VB compilers are fully determinstic when the `/deterministic` option is specified (this is the default in the .NET SDK). This means that the "same inputs" will cause the compilers to produce the "same outputs" byte for byte.
```
Comment (Therzok) [inline docs/compilers/Deterministic Inputs.md:4]:
```suggestion
The C# and VB compilers are fully deterministic when the `/deterministic` option is specified (this is the default in the .NET SDK). This means that the "same inputs" will cause the compilers to produce the "same outputs" byte for byte. 
```

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 4)

**Review by cston — APPROVED** [review]

**Pull request merged**